Modules/Core/Common/include/itkMetaDataObject.hxx

```diff
@@ -28,9 +28,79 @@
 #ifndef itkMetaDataObject_hxx
 #define itkMetaDataObject_hxx
 
+#include <type_traits>
+#include <iterator>
+
+template <typename T, typename = void>
+inline constexpr bool is_iterable_print_v = false;
+// Specialize for std::vector<T> and std::vector<std::vector<T>>
+template <typename T>
+inline constexpr bool is_iterable_print_v<std::vector<T>, std::void_t<>> = true;
+template <typename T>
+inline constexpr bool is_iterable_print_v<std::vector<std::vector<T>>, std::void_t<>> = true;
 
 namespace itk
 {
+template <typename TIterable>
+void
+printIterable(std::ostream & os, const TIterable & iterable)
```

Comment:
Oh, this reminds me of https://github.com/InsightSoftwareConsortium/ITK/issues/513#issue-411127172

Reply:
Maybe Iterable is not the best name here. I started the implementation to be general to any iterable,
but then I realized that there are quite a few iterables that are better printed as one element instead of a "list" of elements (std::string).

At the end I decided to only provide the specialization for `std::vector<T>` and `std::vector<std::vector<T>>` because those two are the only two cases explicitly instantiated that don't have `operator<<` defined.

Reply:
OK. Still saves typing and provides consistency, so would be nice to introduce it across the code base as time permits. If an overload is ever implemented, only one line would need to be substituted for each match in that case.

Reply:
Sorry to chime in again Pablo. Went through the rest of the comments, especially those where Niels proposes this to be a private static method, and I realize that we will not be able to use this in other classes' `PrintSelf` method. So linked to https://github.com/InsightSoftwareConsortium/ITK/pull/4368/files#r1426050481, wouldn't it be better to make this method dwell in a common place where it can be used for that purpose?